public/app/features/query/components/QueryEditorRows.tsx

```diff
@@ -59,10 +58,6 @@ export class QueryEditorRows extends PureComponent<Props> {
   onDataSourceChange(dataSource: DataSourceInstanceSettings, index: number) {
     const { queries, onQueriesChange } = this.props;
 
-    if (this.props.onDatasourceChange) {
-      this.props.onDatasourceChange(dataSource, queries[index]);
-    }
```

Comment:
I think this could potentially still be a valid option, but then it should not also call onQueriesChange, if we have this function the logic that change the query model with the new data source needs to then be moved outside this component. (the lines below that call onQueriesChange needs to be removed and moved to all consumers of this component and implemented in onDatasourceChange)